revpi_hat/hat.py

```diff
@@ -0,0 +1,84 @@
+from datetime import date
+import json
+from typing import List
+
+class RevPiHatEEPROMAttributeException(Exception):
+    pass
+
+
+class RevPiHatEEPROM:
+    def __init__(self, hat_eeprom_path: str = "/proc/device-tree/hat/", load_contents: bool = True) -> None:
+        self._hat_eeprom_path = hat_eeprom_path
+
+        self.uuid: str = None
+        self.format_version: int = None
+
+        self.vendor: str = None
+        self.product: str = None
+        self.product_id: int = None
+        self.product_id_revision: str = None
+        self.product_revision: int = None
+        self.product_version: float = None
+
+        self.serial: int = None
+        self.eol_date: date = None
+        self.batch_number: int = None
+        self.first_mac_address: str = None
+
+        if load_contents:
+            self.load()
+
+    def load(self):
+        """
+        Load values from RevPi HAT EEPROM
+        """
+        self.uuid = self._hat_attribute("uuid")
+        self.format_version = self._hat_attribute_int("custom_0")
+
+        self.vendor = self._hat_attribute("vendor")
+        self.product = self._hat_attribute("product")
+        self.product_id = self._hat_attribute_int("product_id") + 100000
+        self.product_revision = self._hat_attribute_int("custom_2")
+        self.product_version = self._hat_attribute_int("product_ver") / 100
```

Comment:
This is wrong. The version consists of the major and the minor number part. so 101 is 1.1 and 110 is 1.10. See https://github.com/RevolutionPi/revpi-hat-eeprom/blob/master/docs/RevPi-HAT-EEPROM-Format.md#product-version-pver